extensions/schema-compare/src/dialogs/schemaCompareDialog.ts

```diff
@@ -45,18 +46,25 @@ export class SchemaCompareDialog {
 	private sourceIsDacpac: boolean;
 	private targetIsDacpac: boolean;
 	private database: string;
-	public dialogName: string;
+	private connectionId: string;
 
 	protected initializeDialog(): void {
 		this.schemaCompareTab = azdata.window.createTab(SchemaCompareLabel);
 		this.initializeSchemaCompareTab();
 		this.dialog.content = [this.schemaCompareTab];
 	}
 
-	public openDialog(p: any, dialogName?: string): void {
-		let profile = p ? <azdata.IConnectionProfile>p.connectionProfile : undefined;
+	public async openDialog(context: any, dialogName?: string): Promise<void> {
+		let profile = context ? <azdata.IConnectionProfile>context.connectionProfile : undefined;
```

Comment:
tiny check here - its possible that context is not null but isn't IConnectionProfile - in that case the check before ? will succeed but the actual setting will fail. Can you please check for null after casting to IConnectionProfile

Reply:
isn't it ok if it's set to null?